docs/how-to-guides/openstack-swift.md

```diff
@@ -9,26 +9,34 @@ First, connect to your openstack instance with your credentials, sourcing an .rc
 source myfile.rc
 ```
 if you don't have any credentials check [here](https://docs.openstack.org/zh_CN/user-guide/common/cli-set-environment-variables-using-openstack-rc.html) for more information
-Then list all the objects on the openstack containers in debug mode:
+The openstack container you are going to work with has to be globally readable. check that property:
 ```
-openstack object list default -vv
+swift stat <container_name>
 ```
-You will see a debug log as output. The URL should be located in the URL section of a `curl` command. The swift URL will look like this:
+And if it's not globally readable yet, change that with:
 ```
-http://10.126.72.107:8080/v1/AUTH_fa8326b9fd4f405fb1c5eaafe988f5fd/default
+swift post <container_name> --read-acl ".r:*,.rlistings"
 ```
-Or directly filter the url from the debug log instead of searching into the output:
+Then list all the objects on the openstack container in debug mode and get the url of the swift connection:
 ```
-openstack object list default -vv 2>&1 | grep "^REQ: curl" | grep AUTH | cut -d'"' -f2 | cut -d'?' -f1
+openstack object list <container_name> -vv 2>&1 | grep "^REQ: curl" | grep AUTH | cut -d'"' -f2 | cut -d'?' -f1
 ```
-After obtaining the URL, configure the charm to use that url as backend and site. Assuming the charm has been deployed as content-cache-k8s, run:
+After obtaining the URL, configure the charm to use that url as our charm's backend. Assuming the charm has been deployed as content-cache-k8s, run:
 ```
-juju config content-cache-k8s backend="http://<swift_conn_url>:80"
-juju config content-cache-k8s site="<swift_conn_url>"
+juju config content-cache-k8s backend="http://<swift_conn_url>"
```

Comment:
I think that would just be `backend="<swift_conn_url>"` as the URL will already include http or https. Actually, from looking at the integration tests we use `f"http://{swift_conn.url}:80"}`. So I'm not sure we want to include the full output of the `openstack object list` command, as that's got a path, not just a hostname. Can we confirm what's needed, and how this would work with http and https?